app/repositories/pagerepository.php

```diff
@@ -286,6 +288,149 @@ public function getAllChildPageLinks()
         }
     }
 
+    /**
+     * Deletes a page and its related entities by id
+     * @param int $id
+     * @return void
+     * @throws \Exception
+     * @author Luko Pecotic
+     */
+    public function deletePage($id)
+    {
+        try {
+            $this->connection->beginTransaction();
+
+            $this->deleteInfoTextsByPageId($id);
+            $this->deleteCardsByPageId($id);
```

Comment:
dit kan netter, ik weet niet zeker hoe het nu met de database is maar daat kan je on delete cascade doen, dan delete hij ook alle gerelateerde dingen